src/new.rs

```diff
@@ -88,6 +89,30 @@ pub fn create_new_project<P: AsRef<Path>>(dest: P) -> Result<()> {
     Ok(())
 }
 
+pub fn create_new_post(name: &String, config: &Config) -> Result<()> {
+    let path = Path::new(&config.source);
+    let full_path = &path.join(&config.posts).join(name);
+
+    try!(create_file(full_path, post_1_md));
+
+    Ok(())
+}
+
+pub fn create_new_layout(name: &String, config: &Config) -> Result<()> {
+    let path = Path::new(&config.source);
+    let full_path = &path.join(&config.layouts).join(name);
+
+    try!(create_file(full_path, post_liquid));
+
+    Ok(())
+}
+
+pub fn create_new_page(name: &String) -> Result<()> {
+    try!(create_file(name, index_liquid));
+
+    Ok(())
+}
```

Comment:
Also would be worth stubbing out the new file. I ended up adding something like this

```
const new_post_tmpl: &'static [u8] = b"extends: default.liquid

title: %title%
date: %date%
---
";

const new_page_tmpl: &'static [u8] = b"extends: default.liquid

title: %title%
route: %title%
---
";

pub fn create_post<P: AsRef<Path>>(dest: P, title: &str) -> Result<()> {
  let dest = dest.as_ref();
  // 7 may 2017 21:03:34 +0000
  let local = Local::now().format("%d %b %Y %H:%M:%S %z").to_string();
  let vector: Vec<u8> = Vec::from(new_post_tmpl);
  let mut out = String::from_utf8(vector).unwrap();
  out = out.replace("%title%", title);
  out = out.replace("%date%", &local);
  let new_post = &dest.join(format!("posts/{}.md", title));
  try!(create_file(new_post, out.as_bytes()));
  Ok(())
}
```

which basically adds the title and time stamp to the new post or page.